api.go

```diff
@@ -30,14 +30,21 @@ type apiResponse struct {
 	Payload *json.RawMessage
 }
 
-func (s *Session) apiRequest(endpoint string, mixins []string) (*json.RawMessage, error) {
-	theurl := s.baseurl
-	params := url.Values{
+// apiRequestWithParams conducts a GET request with custom parameters.
+func (s *Session) apiRequestWithParams(endpoint string, mixins []string, params map[string][]string) (*json.RawMessage, error) {
```

Comment:
That's the underlying type of `url.Values`, so I thought I'd follow suit for maximum generality.